### DISCUSSION TITLE
Migrating from old Nora to Smart Nora on Raspberry Pi B

### DISCUSSION THREAD

**Discussion by 420pootang69** [andrei-tatar/node-red-contrib-smartnora#250] [General]
Hi,

So I used the old Nora for ages before it became unstable, I've been looking forward to trying the new smart Nora, but only had the time this weekend to set it up.

I installed with "npm install node-red-contrib-smartnora", which showed a dependency was missing. I fixed this with (I think) "npm I encoding". Now all is installed, but it doesn't show up in Node red. It also still gives me the option to install from within Node Red, but this won't work as npm and Node red running at the same time will kill the Pi due to the high RAM usage that Smart Nora uses while installing (Why is this anyway? never seen this before). 

I've been at this for two days now! Any help appreciated. I've an elderly parent that relied on the old Nora and it was sorely missed.

Thanks.

**Comment by andrei-tatar** [comment]
Hi,

What versions are you running for `node.js` / `node-red` / `npm`?
I haven't tried running this on a RPi 1. Is it the one with 512 MB of RAM? 
Are you running the `Lite` version of the OS?

**Reply by 420pootang69** [reply]
Hi!

It's Raspbian that I downloaded from the pi website when I got the device. It's 512MB yes. 
I'm running the latest version of all 3 that this Pi supports, however they are quite old.

Node-RED version: v0.20.7
Node.js  version: v11.15.0
Linux 4.19.66+ arm LE
npm version: 6.7.0

Thanks for the quick reply!

**Reply by andrei-tatar** [reply]
@420pootang69 Sorry but smartnora only supports node.js >= 12.19. Isn't there node.js 12 for RPi 1?

**Reply by 420pootang69** [reply]
Ah no...

"apt install nodejs" tells me I'm on the latest version.

I then checked "https://nodejs.org/dist/" and I can see an ARM61 build for V11, but not for version 12.

Assuming I'm correct, can you maybe put a note somewhere on the Node red page that the original Pi isn't supported? 

Thanks for your quick replies anyway, pity there wasn't a solution.

**Reply by andrei-tatar** [reply]
Have you tried running this script directly?
https://nodered.org/docs/getting-started/raspberrypi

**Reply by 420pootang69** [reply]
Ignore this for a second, sorry.

**Reply by 420pootang69** [reply]
Sorry, my original comment was correct.
The last version officially supported for the Pi was V8, so that's what the script downloads. Never mind the fact node red won't work with this version. I had to download version 11 manually.
I've done so much with this over the last two days I'm getting confused! I've never looked at node.js until two days ago and even the version numbering is still confusing to me!

**Reply by 420pootang69** [reply]
https://i.ibb.co/YNKSBkw/image.png

Just to show that it won't upgrade past version V11.

**Reply by andrei-tatar** [reply]
Hmm. Honestly I would suggest switching to something a bit more modern. Node.js is already at v18 and RPi 1 is more than 10 years old already.

**Comment by 420pootang69** [comment]
Have you seen the cost of Pis currently! :D 

Anyway. Would you please leave a note on the Node Red page to let people know about this issue so they don't spend a weekend on it like I did?

For my dad, I'm going to come up with a different solution. I am setting up a similar setup for myself though and it will be running on a PC, so I will be using smart Nora in the future :)

**Comment by 420pootang69** [comment]
I actually got a lot further with this. While there is no official build of V12 of nodejs for ARMV61, there is an unofficial build:
https://unofficial-builds.nodejs.org/

I managed to get version 12 loaded and working with Node red perfectly. I then went to install smart Nora and watched the log. It fully downloaded, went to the post install script, and then threw an error (trying to think from memory)  code 1...

Any ideas?

**Reply by andrei-tatar** [reply]
Can you post the logs?

**Reply by 420pootang69** [reply]
Hi Andrei,

Unfortunately I couldn't afford to work on this anymore after today. I installed a WiFi wall switch to allow my father to turn all the lights on and off, which should resolve his issue.

Sorry I couldn't do more testing on this. It appears it will work, there is just some manual steps involved.

If I do get a chance to try this again, what logs exactly would you need? The one in the Node Red interface just shows error code 1 after the install script, so I assume there is a more detailed log somewhere?

**Reply by andrei-tatar** [reply]
@420pootang69 you can manually install it using npm and copy the logs

```sh
cd ~/.node-red
npm install node-red-contrib-smartnora@latest
```
